libr/anal/p/anal_mips_gnu.c

```diff
@@ -37,13 +38,23 @@ static int mips_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *b, int len, R
 	// Be endian aware
 	opcode = r_read_ble32 (b, anal->big_endian);
 
-	// eprintf ("MIPS: %02x %02x %02x %02x (after endian: big=%d)\n", b[0], b[1], b[2], b[3], anal->big_endian);
+	// eprintf ("MIPS: %02x %02x %02x %02x (after endian: big=%d)\n", buf[0], buf[1], buf[2], buf[3], anal->big_endian);
 	if (opcode == 0) {
 		op->type = R_ANAL_OP_TYPE_NOP;
 		return oplen;
 	}
-
-	optype = (b[0]>>2);
+	
+	if (anal->big_endian) {
+		memcpy(buf, b, sizeof(ut8)*4);	
```

Comment:
please use just `4` or `sizeof(ut32)`

Reply:
ok
This is another pull request, I made a mistake, sorry